aiohttp/client.py

```diff
@@ -543,7 +543,10 @@ async def _request(
 
                     if auth is None:
                         auth = auth_from_url
-                    if auth is None:
+
+                    if auth is None and (
+                        not self._base_url or self._base_url.origin() == url.origin()
```

Comment:
Calling `.origin()` is a bit expensive https://github.com/aio-libs/aiohttp/issues/7583#issuecomment-1806828707

We should probably cache `self._base_url.origin()` as self._base_url_origin` so we don't have to build it every time

Reply:
Probably could guard this with `self._default_auth` being set as well

Reply:
https://github.com/aio-libs/aiohttp/pull/9466